src/components/CustomSelect.jsx

```diff
@@ -24,37 +27,72 @@ function CustomSelect({ options, selected, onSelect }) {
     };
   });
 
+  useEffect(() => {
+    const liHeight = liRefs.current[focusedIndex]?.offsetHeight;  //total height of li element
+    const scrollTop = ulRef.current?.scrollTop;                   //shows the scroll position of ul element
+    const viewport = scrollTop + ulRef.current?.offsetHeight;     //sum of scroll position of ul and it's total height
+    const liOffset = liHeight * focusedIndex;                     //total height of li element depending on the index
+    
+    console.log('focusedIndex', focusedIndex, ' viewport', viewport, ' liOffset', liOffset);
+
+    if (ulRef.current && (liOffset < scrollTop || liOffset + liHeight > viewport)) {
+      ulRef.current.scrollTop = liOffset;
```

Comment:
I like it :) it is simple and safe solution to scroll to current element 💪 
what I might suggest is to make select list quite bigger to user might see the part of next element of the list to tell that there is more options to see 😄

Reply:
✅